### PR TITLE
From Pipenv to PDM

### PR DESCRIPTION
**Pull request by SharonHart**
## Change Description

* Drop pipenv for PDM
* Implement DRY by instructing PDM to take dependencies from the new pyproject.toml, dropping the Pipfiles. 
* That change required moving server dependencies to the pyproject.toml, as 'server' optional dependencies. Same goes for dev dependencies.

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by omri374** [comment]
Thanks @SharonHart I like the DRY approach, but it looks like we're doing a detour instead of just using Poetry which directly uses the pyproject.toml file. Maybe we can test poetry and see if it works well for us? If it doesn't fit our needs for some reason, I agree that this PR is the right approach.
WDYT?

**Comment by SharonHart** [comment]
> Thanks @SharonHart I like the DRY approach, but it looks like we're doing a detour instead of just using Poetry which directly uses the pyproject.toml file. Maybe we can test poetry and see if it works well for us? If it doesn't fit our needs for some reason, I agree that this PR is the right approach. WDYT?

Yes, I this is the intermediate step, I thought to introduce poetry later on, and in stages since I think it would mean migrating from setuptools as well

**Comment by omri374** [comment]
Ok sounds good. Please update https://github.com/microsoft/presidio/blob/main/docs/development.md too

File: presidio-analyzer/Dockerfile [hunk 1]
```diff
@@ -17,14 +17,14 @@ COPY ${NLP_CONF_FILE} /usr/bin/${NAME}/${NLP_CONF_FILE}
...
+COPY ./pyproject.toml /usr/bin/${NAME}/
 COPY ./Pipfile* /usr/bin/${NAME}/
 
 # Install essential build tools
 RUN apt-get update \
   && apt-get install -y build-essential
 
-RUN pip install pipenv \
-  && pipenv install --deploy
+RUN pip install pipenv && pipenv install pipenv install -e '.[server,stanza,azure-ai-language]'
```
Comment (omri374) [inline presidio-analyzer/Dockerfile:27]: Should the other Dockerfiles be updated too?

Comment (SharonHart) [reply presidio-analyzer/Dockerfile:27]: Yes, I was POCing with analyzer

**Comment by SharonHart** [comment]
> Ok sounds good. Please update https://github.com/microsoft/presidio/blob/main/docs/development.md too

I am also considering pdm

File: docs/development.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ Follow these steps when starting to work on a Presidio service with Pipenv:
     For example, in the `presidio-analyzer` folder, run:
 
     ```
-    pipenv install --dev --skip-lock
+    pipenv install -e '.[all]' --skip-lock
```
Comment (omri374) [inline docs/development.md:48]: pdm

**Comment by SharonHart** [comment]
Closed in favor of https://github.com/microsoft/presidio/pull/1391

**Comment by SharonHart** [comment]
Closed in favor of https://github.com/microsoft/presidio/pull/1391

**Pull request closed without merging**